File: WebAPI/IAI.Models/Models/Candidate/Requests/CandidateProfileRequest.cs

using System.ComponentModel.DataAnnotations;

namespace IAI.Models.Models.Candidate.Requests
{
    public class CandidateProfileRequest
    {
        [Required]
        public Guid CandidateId { get; set; }
        [Required]
        public int CityId { get; set; }
        [Required]
        public int ExperienceId { get; set; }
        [Required]
        public int DesignationId { get; set; }
        [Required]
        public decimal CurrentPay { get; set; }
        public int NoticePeriodId { get; set; }
        [Required]
        public long MobileNumber { get; set; }
        [Required]
        public string? EmailId { get; set; }
        [Required]
        public bool RestrictEmployerToViewProfile { get; set; }
    }
}

File: WebApp/iai-app/src/models/candidate/request/CandidateProfileRequest.ts
interface CandidateProfileRequest {
    candidateId: string;
    cityId: number;
    experienceId: number;
    designationId: number;
    currentPay: number;
    noticePeriodId: number;
    mobileNumber: number;
    emailId: string;
    resumeTitle?: string;
    restrictEmployerToViewProfile?: boolean;
}
export default CandidateProfileRequest;